src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -107,9 +107,7 @@ public IEnumerable<DiagnosticData> ConvertToLocalDiagnostics(Document targetDocu
                         var version = await GetDiagnosticVersionAsync(project, cancellationToken).ConfigureAwait(false);
                         var existingData = await ProjectAnalysisData.CreateAsync(project, stateSets, avoidLoadingData, cancellationToken).ConfigureAwait(false);
 
-                        // we can't return here if we have open file only analyzers sine saved data for open file only analyzer
-                        // is wrong. (since it only contains info on open files rather than whole project)
```

Comment:
you should do something here I believe. otherwise, fix all on open file only analyzer will be broken. not sure whether there is a test cover this or not though.

Reply:
I see why you removed it here now.